ledger/metrics/src/test/suite/scala/com/daml/metrics/grpc/GrpcMetricsServerInterceptorSpec.scala

```diff
@@ -0,0 +1,147 @@
+    extends AsyncFlatSpec
+    with Matchers
+    with AkkaBeforeAndAfterAll
+    with TestResourceContext
+    with MetricValues {
+
+  private val labelsForSimpleRequest = MetricsContext(
+    Map(
+      "grpc_client_type" -> "unary",
+      "grpc_service_name" -> "com.daml.platform.HelloService",
+      "grpc_method_name" -> "Single",
+      "grpc_server_type" -> "unary",
+    )
+  )
+
+  private val labelsForSimpleRequestWithStatusCode = labelsForSimpleRequest.merge(
+    MetricsContext(
+      Map(
+        "grpc_code" -> "OK"
+      )
+    )
+  )
+
+  behavior of "Grpc server metrics interceptor"
+
+  it should "time call" in {
+    val metrics = new TestingGrpcMetrics
+    withService(metrics).use { helloService =>
+      helloService.single(new HelloRequest()).map { _ =>
+        metrics.callTimer.getCounts should contain theSameElementsAs Map(
+          labelsForSimpleRequestWithStatusCode -> 1
+        )
+      }
+    }
+  }
+
+  it should "mark call start and finish" in {
+    val metrics = new TestingGrpcMetrics
+    withService(metrics).use { helloService =>
+      helloService.single(new HelloRequest()).map { _ =>
+        metrics.callsStarted.valueWithContext should contain theSameElementsAs Map(
+          labelsForSimpleRequest -> 1
+        )
+        metrics.callsFinished.valueWithContext should contain theSameElementsAs Map(
+          labelsForSimpleRequestWithStatusCode -> 1
+        )
+      }
+    }
+  }
+
+  it should "mark streaming messages" in {
+    val metrics = new TestingGrpcMetrics
+    withService(metrics).use { helloService =>
+      val streamConsumer = new StreamConsumer[HelloResponse](observer =>
+        helloService.serverStreaming(HelloRequest(3), observer)
+      )
+      streamConsumer.all().map { _ =>
+        def meterHasValueForStreaming(
+            meter: MetricHandle.Meter,
+            metricsContext: MetricsContext = withStreamingLabels(labelsForSimpleRequest),
+            value: Long = 1,
+        ) = {
+          meter.valueWithContext should contain theSameElementsAs Map(
+            metricsContext -> value
+          )
+        }
+        meterHasValueForStreaming(metrics.callsStarted)
+        meterHasValueForStreaming(
+          metrics.callsFinished,
+          withStreamingLabels(labelsForSimpleRequestWithStatusCode),
+        )
+        meterHasValueForStreaming(metrics.messagesSent, value = 3)
+        meterHasValueForStreaming(metrics.messagesReceived)
+        meterHasValueForStreaming(metrics.messagesRequested, value = 2)
+      }
+    }
+  }
+
+  it should "monitor sizes of messages" in {
+    val metrics = new TestingGrpcMetrics
+    withService(metrics).use { helloService =>
+      val payload = ByteString.copyFromUtf8("test message")
+      helloService.single(new HelloRequest(payload = payload)).map { _ =>
+        metrics.messagesSentSize.valuesWithContext should contain theSameElementsAs Map(
+          labelsForSimpleRequest -> Seq(14)
```

Comment:
How did we get a size of 14 here? I'd suggest spelling out the calculation of this value (e.g., by getting the serialized size from `HelloRequest` or `HelloResponse`).